### PR TITLE
Restore `setCreator` API dropped by accident in a minor release

### LINKED ISSUE #286: v4.28.0 broken public api
I don't know if its intentional, but https://github.com/openspout/openspout/releases/tag/v4.28.0 removed public setCreator method from the XLSX Writer as part of https://github.com/openspout/openspout/pull/285/files and it seems very odd that this is done on a minor release ?
Or I've missed something in the recent updates

i would keep the setCreator method and simply make it update the underlying properties object (it's readonly, so it's a bit tricky if you keep it that way)

### PR DESCRIPTION
**Pull request by lekoala**
this will also require Properties to not be readonly

fixes https://github.com/openspout/openspout/issues/286

### CODE REVIEW

File: src/Writer/XLSX/Properties.php [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 namespace OpenSpout\Writer\XLSX;
 
-final readonly class Properties
+final class Properties
```
Comment (Slamdunk) [inline src/Writer/XLSX/Properties.php]: This is not needed, just overwrite the object with a new one

File: src/Writer/XLSX/Writer.php [hunk 2]
```diff
@@ -35,6 +35,11 @@ public function getOptions(): Options
         return $this->options;
     }
 
+    public function setCreator(string $creator): void
```
Comment (Slamdunk) [inline src/Writer/XLSX/Writer.php:38]: I guess we also need to restore this in the `WriterInterface`

**Review by Slamdunk — CHANGES_REQUESTED** [review]

**Comment by lekoala** [comment]
@Slamdunk kept it readonly (it's just a bit verbose imho), added back method to interface. instead of implementing in abstract, i kept an abstract method and added a noop function to the csv writer

File: src/Writer/AbstractWriter.php [hunk 3]
```diff
@@ -137,6 +137,8 @@ final public function close(): void
         $this->isWriterOpened = false;
     }
 
+    abstract public function setCreator(string $creator): void;
```
Comment (Slamdunk) [inline src/Writer/AbstractWriter.php]: This is already enforced by the interface and can be dropped here

File: src/Writer/CSV/Writer.php [hunk 4]
```diff
@@ -28,6 +28,11 @@ public function getOptions(): Options
     {
         return $this->options;
     }
+    
+    public function setCreator(string $creator): void
+    {
+        // noop
```
Comment (Slamdunk) [inline src/Writer/CSV/Writer.php]: May it would be better to raise an Exception message like `Method unsopported for CSV documents`

**Review by Slamdunk — CHANGES_REQUESTED** [review]

**Comment by lekoala** [comment]
@Slamdunk i've made the requested changes

i'm not planning on updating this pr further than that, it feels like you would spend as much time writing the comment as doing the change yourself which seem counterproductive. my proposal would be to merge as is and make any last changes as you see fit.

have a great day,

**Comment by Slamdunk** [comment]
> it feels like you would spend as much time writing the comment as doing the change yourself which seem counterproductive

It depends on the definition of counterproductive.
I review PR on the open source projects I maintain to share my knowledge so that the knowledge itself can improve for everyone from anyone has better ideas and suggestions.

If you feel annoyed by my feedbacks, imagine how I feel having to deal with hundreds of people like you after I spent my free time in the pursuit of reciprocal help.

**Review by Slamdunk — APPROVED** [review]

**Pull request merged**